Fix getParentsPosts() query

asubset opened the pull request:
DISREGARD this comment-
postgres DISTINCT includes empty/null columns - the getparentsposts query was essentially returning all posts (for postgres) with no rootId, assuming one post row within the limit/offset had an empty rootId

jwilander commented:
Hi @asubset , thanks for the pull request!

Before we can review and accept your PR we'll need you to sign the CLA here http://www.mattermost.org/mattermost-contributor-agreement/ .

crspeller commented:
@asubset This also fails some unit tests. See the travis output for details.

coreyhulen commented:
Also we support mysql so I think <> needs to be !=

asubset commented:
My apologies for the premature pull req.

What I thought was a difference in postgres was not. But I do not think getParentPosts() is behaving as intended.

Bug 1: It is selecting parents based on a child.RootId = parent.RootId join. I believe this should be child.RootId = parent.Id.

Bug 2: It is also returning all posts with an empty RootId - therefore typically making up for bug 1, with some overkill.

I've changed the join to RootId = Id and excluded empty RootIds from the subquery.

I think I have this right now. Again sorry for submitting too early.

--
-I've submitted the CLA
-I believe != and <> are synonymous in both MySql and Postgres. But I changed to != for consistency with that usage elsewhere in the code.

coreyhulen commented:
@asubset good catch.

jwilander commented:
+1
